pkg/metric/writer_factory.go

```diff
@@ -9,20 +9,34 @@ import (
 )
 
 const (
-	WriterTypeCw   = "cw"
-	WriterTypeES   = "es"
-	WriterTypeProm = "prom"
+	WriterTypeCloudwatch    = "cw"
+	WriterTypeElasticsearch = "es"
+	WriterTypePrometheus    = "prom"
 )
 
-func ProvideMetricWriterByType(ctx context.Context, config cfg.Config, logger log.Logger, typ string) (Writer, error) {
-	switch typ {
-	case WriterTypeCw:
-		return NewCwWriter(ctx, config, logger)
-	case WriterTypeES:
-		return NewEsWriter(config, logger)
-	case WriterTypeProm:
-		return NewPromWriter(ctx, config, logger)
+func ProvideMetricWriter(ctx context.Context, config cfg.Config, logger log.Logger, types []string) (Writer, error) {
```

Comment:
this is not a real `Provide*` in our usual sense, isn't it?\
This is more like the normal `NewMultiWriter`?